src/Sylius/Bundle/CoreBundle/Checkout/Step/FinalizeStep.php

```diff
@@ -66,6 +68,7 @@ protected function completeOrder(OrderInterface $order)
         $this->dispatchCheckoutEvent(SyliusOrderEvents::PRE_CREATE, $order);
         $this->dispatchCheckoutEvent(SyliusCheckoutEvents::FINALIZE_PRE_COMPLETE, $order);
 
+        $this->applyTransition('complete', $order);
         $this->get('sm.factory')->get($order, OrderTransitions::GRAPH)->apply(OrderTransitions::SYLIUS_CREATE, true);
```

Comment:
It's because there are two different state machines here - ``applyTransition`` function applies transitions from ``sylius_order_checkout`` state machine, this line however runs transition from ``sylius_order``.
I don't know should we create another function (or pass graph name as parameter) only for this single call.